stock_picking_invoice_link/stock.py

```diff
@@ -18,69 +18,59 @@
 
-    _columns = {
-        'invoice_line_id': fields.many2one(
-            'account.invoice.line', 'Invoice Line', readonly=True),
-    }
+    invoice_line_id = fields.Many2one(comodel_name='account.invoice.line',
+                                      string='Invoice Line', readonly=True)
 
-    def _create_invoice_line_from_vals(
-            self, cr, uid, move, invoice_line_vals, context=None):
-        inv_line_id = super(stock_move, self)._create_invoice_line_from_vals(
-            cr, uid, move, invoice_line_vals, context=context)
-        move.write({'invoice_line_id': inv_line_id})
+    @api.model
+    def _create_invoice_line_from_vals(self, move, invoice_line_vals):
+        inv_line_id = super(StockMove, self)._create_invoice_line_from_vals(
+            move, invoice_line_vals)
+        move.invoice_line_id = inv_line_id
         move.picking_id.invoice_id = invoice_line_vals['invoice_id']
         return inv_line_id
 
 
-class stock_picking(orm.Model):
+class StockPicking(models.Model):
     _inherit = "stock.picking"
 
-    def _get_invoice_view_xmlid(self, cr, uid, ids, name, arg, context=None):
-        res = {}
-        for pick in self.browse(cr, uid, ids, context=context):
+    @api.multi
+    def _get_invoice_view_xmlid(self):
+        for pick in self:
             if pick.invoice_id:
                 if pick.invoice_id.type in ('in_invoice', 'in_refund'):
-                    res[pick.id] = 'account.invoice_supplier_form'
+                    pick.invoice_view_xmlid = 'account.invoice_supplier_form'
                 else:
-                    res[pick.id] = 'account.invoice_form'
-            else:
-                res[pick.id] = False
-        return res
+                    pick.invoice_view_xmlid = 'account.invoice_form'
 
-    _columns = {
-        'invoice_id': fields.many2one(
-            'account.invoice', 'Invoice', readonly=True),
-        'invoice_view_xmlid': fields.function(
-            _get_invoice_view_xmlid, type='char', string="Invoice View XMLID",
-            readonly=True),
-    }
+    invoice_id = fields.Many2one(comodel_name='account.invoice',
+                                 string='Invoice', readonly=True)
+    invoice_view_xmlid = fields.Char(compute='_get_invoice_view_xmlid',
+                                     string="Invoice View XMLID",
+                                     readonly=True)
 
 
-class account_invoice(orm.Model):
+class AccountInvoice(models.Model):
     _inherit = "account.invoice"
 
-    _columns = {
-        'picking_ids': fields.one2many(
-            'stock.picking', 'invoice_id', 'Related Pickings', readonly=True,
-            copy=False,
-            help="Related pickings "
-            "(only when the invoice has been generated from the picking)."),
-    }
+    picking_ids = fields.One2many(
+        comodel_name='stock.picking', inverse_name='invoice_id',
+        string='Related Pickings', readonly=True,
+        copy=False,
+        help="Related pickings "
```

Comment:
Me too usually! So https://github.com/tafaRU/stock-logistics-workflow/commit/57e6ac2c40108fe4f61b180a7ad3f32f49c8beae :wink: 

Does the pot file need to be exported again?